### ISSUE TITLE
Bug: DEFAULT_FROM_EMAIL silently becomes empty string when EMAIL_HOST_USER is not set

### ISSUE THREAD

**Issue by 5hivam123** [Checkora/Checkora#1886]
## Description
When neither `DEFAULT_FROM_EMAIL` nor `EMAIL_HOST_USER` is set in the `.env` 
file, `DEFAULT_FROM_EMAIL` in `core/settings.py` silently resolves to an 
empty string `""`. This causes all outgoing emails (OTP, password reset, 
welcome email) to fail silently with no error or warning raised by Django.

## Steps to Reproduce
1. Clone the repository and set up the project locally
2. In `.env`, do NOT set `EMAIL_HOST_USER` or `DEFAULT_FROM_EMAIL`
3. Register a new account to trigger an OTP/welcome email
4. See that email fails silently — no error shown, no Django exception raised

## Expected Behavior
Django should raise an `ImproperlyConfigured` error or at minimum a 
`warnings.warn()` alert if `DEFAULT_FROM_EMAIL` resolves to an empty string, 
so developers are immediately aware of the misconfiguration.

## Actual Behavior
`DEFAULT_FROM_EMAIL` becomes `""` silently. Emails are dispatched with a 
blank `From:` address, which SMTP servers reject, but the application shows 
no feedback to the developer.

## Screenshots
N/A — this is a silent configuration bug with no visible UI error.

## Environment
- OS: Windows 11
- Browser: N/A (backend/settings issue)
- Python Version: 3.12

## Additional Context
Affected file: `core/settings.py`
Relevant line:
```python
DEFAULT_FROM_EMAIL = os.getenv('DEFAULT_FROM_EMAIL') or EMAIL_HOST_USER
```
Fix suggestion — add after that line:
```python
if not DEFAULT_FROM_EMAIL:
    raise ImproperlyConfigured(
        "DEFAULT_FROM_EMAIL is not configured. Set EMAIL_HOST_USER "
        "or DEFAULT_FROM_EMAIL in your .env file."
    )
```
Labels suggested: `bug`, `good first issue`, `GSSoC`

**Comment by 5hivam123** [comment]
I'd like to work on this fix myself as part of GSSoC. Can you please assign it to me? 🙏 #GSSoC

**Comment by saurabhhhcodes** [comment]
Hi @Checkora/maintainers, I've made the fix for this issue. Could I please be assigned so I can open a PR? Thanks!

**Comment by saurabhhhcodes** [comment]
Hello! I'd like to work on this issue under GSSoC 26. Please assign it to me so I can submit a fix. Thanks!
